lib/internal/util/iterable-weak-map.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const {
+  Set,
+  SetPrototypeAdd,
+  SetPrototypeDelete,
+  Symbol,
+  WeakMap,
+  WeakMapPrototypeGet,
+  WeakMapPrototypeDelete,
+  WeakMapPrototypeSet,
```

Comment:
I think you can simplify this quite a lot by using `SafeWeakMap` and `SafeSet` instead:

```suggestion
  SafeSet,
  SafeWeakMap,
  Symbol,
```